src/main/java/me/xmrvizzy/skyblocker/skyblock/item/PriceInfoTooltip.java

```diff
@@ -20,105 +22,130 @@
-                    || (!listString.contains("Lowest BIN Price") && lowestPricesJson != null && lowestPricesJson.has(name))) {
+            if ((!listString.contains("Avg. BIN Price") && (threeDayAvgPricesJson.has(name) || oneDayAvgPricesJson.has(name)))
+                    || (!listString.contains("Lowest BIN Price") && lowestPricesJson.has(name))) {
 
-                if (SkyblockerConfig.get().general.itemTooltip.enableLowestBIN) {
+                if (SkyblockerConfig.get().general.itemTooltip.enableLowestBIN && lowestPricesJson.has(name)) {
                     lines.add(new LiteralText(String.format("%-19s", "Lowest BIN Price:"))
                             .formatted(Formatting.GOLD)
                             .append(getCoinsMessage(lowestPricesJson.get(name).getAsDouble(), count)));
                 }
 
-                // Change format from Skytils to Moulberry's for Avg. BIN
                 if (name.contains("PET-")) {
                     name = name.replace("PET-", "")
-                            .replace("0", "COMMON")
-                            .replace("1", "UNCOMMON")
-                            .replace("2", "RARE")
-                            .replace("3", "EPIC")
-                            .replace("4", "LEGENDARY")
-                            .replace("5", "MYTHIC")
+                            .replace("COMMON", "0")
+                            .replace("UNCOMMON", "1")
+                            .replace("RARE", "2")
+                            .replace("EPIC", "3")
+                            .replace("LEGENDARY", "4")
+                            .replace("MYTHIC", "5")
                             .replace("-", ";");
                 } else if (name.contains("ENCHANTED_BOOK-")) {
                     name = name.replace("ENCHANTED_BOOK-", "").replace("-", ";");
+                } else if (name.contains("POTION-")) {
+                    //name = "POTION";
+                    name = "";
+                } else if (name.contains("RUNE-")) {
+                    //name = "RUNE";
+                    name = "";
                 } else {
                     name = name.replace(":", "-");
                 }
 
-                // has(name) check because Skytils keeps old data but Moulberry not
-                if (SkyblockerConfig.get().general.itemTooltip.enableAvgBIN && avgPricesJson.has(name)) {
-                    lines.add(new LiteralText(String.format("%-21s", "Avg. BIN Price:"))
-                            .formatted(Formatting.GOLD)
-                            .append(getCoinsMessage(avgPricesJson.get(name).getAsDouble(), count)));
+                if (SkyblockerConfig.get().general.itemTooltip.enableAvgBIN && !name.isEmpty()
+                        && (threeDayAvgPricesJson.has(name) || oneDayAvgPricesJson.has(name))) {
+                    switch (SkyblockerConfig.get().general.itemTooltip.avg) {
```

Comment:
Did you test if this code works as intended?
Pretty sure that if you set the tooltip to display one day average it will not only display both, it will do so twice. If switch doesn't have breaks, it executes code in all cases after the one it entered.

Also, code duplication. Both problems can be resolved by replacing this switch-case with 2 ifs.